examples/gno.land/p/demo/memeland/memeland.gno

```diff
@@ -0,0 +1,196 @@
+package memeland
+
+import (
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/seqid"
+	"std"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Post - as previously defined
+type Post struct {
+	ID            string
+	Data          string
+	Author        std.Address
+	Timestamp     time.Time
+	Upvotes       int
+	UpvoteTracker *avl.Tree // address > bool hasVoted
+}
+
+type Memeland struct {
+	Posts       *avl.Tree
+	MemeCounter seqid.ID
+}
+
+func NewMemeland() *Memeland {
+	return &Memeland{
+		Posts: avl.NewTree(),
+	}
+}
+
+// PostMeme - Adds a new post
+func (m *Memeland) PostMeme(data, timestamp string) string {
+	if data == "" || timestamp == "" {
+		return "timestamp and data cannot be empty"
+	}
+
+	// Generate ID
+	id := m.MemeCounter.Next().String()
+
+	// Parse unix timestamp into int
+	t, err := strconv.Atoi(timestamp)
+	if err != nil {
+		return "cannot parse timestamp"
+	}
+
+	newPost := &Post{
+		ID:            id,
+		Data:          data,
+		Author:        std.GetOrigCaller(),
+		Timestamp:     time.Unix(int64(t), 0),
+		Upvotes:       0,
+		UpvoteTracker: avl.NewTree(),
+	}
+
+	m.Posts.Set(id, newPost)
+
+	return id
+}
+
+func (m *Memeland) Upvote(id string) string {
+	post, exists := m.Posts.Get(id)
+	if !exists {
+		return "post with specified ID does not exist"
+	}
+
+	p, ok := post.(*Post)
+	if !ok {
+		return "error casting post"
+	}
+
+	caller := std.GetOrigCaller().String()
+
+	if _, exists := p.UpvoteTracker.Get(caller); exists {
+		return "user has already upvoted this post"
+	}
+
+	p.Upvotes += 1
+	m.Posts.Set(id, p)
+	p.UpvoteTracker.Set(caller, true)
+
+	return "upvote successful"
+}
+
+// GetPost - Retrieves a post
+func (m *Memeland) GetPost(id string) (*Post, bool) {
+	value, exists := m.Posts.Get(id)
+	if exists {
+		return value.(*Post), true
+	}
+	return nil, false
+}
+
+func (m *Memeland) GetPosts() []*Post {
+	var posts []*Post
+
+	m.Posts.Iterate("", "", func(key string, value interface{}) bool {
+		post := value.(*Post)
+		posts = append(posts, post)
+
+		return false
+	})
+
+	return posts
+}
+
+// GetPostsByTimestamp returns a JSON string of posts within the given timestamp range, supporting pagination.
+func (m *Memeland) GetPostsByTimestamp(startTimestamp, endTimestamp string, page, pageSize int) string {
```

Comment:
Why isn't the timestamp unix?

Reply:
[71c1f7d](https://github.com/gnolang/gno/pull/1751/commits/71c1f7db7bda97d1a3d2bd5958db5c3db2bc817d)